campaigns: Add support for BitbucketServer build status web hooks (#9371)
sourcegraph/sourcegraph: doc/admin/external_service/bitbucket_server.md

Using webhooks is highly recommended when using [Campaigns](../../user/campaigns.md), since they speed up the syncing of pull request data between Bitbucket Server and Sourcegraph and make it more efficient.

> *NOTE*: Version 1.3.3 or higher is required for Build Status support.

Reviewer: This creates an information box:
```suggestion
> NOTE: Version 1.3.3 or higher is required for Build Status support.
```

> > NOTE: Version 1.3.3 or higher is required for Build Status support.

Reviewer 2: Version 1.3.3. of what? This seems a bit lost here. How about moving it to the documentation for the plugin itself? `../../integration/bitbucket_server.md#sourcegraph-bitbucket-server-plugin` section.

To set up webhooks:

1. Connect Bitbucket Server to Sourcegraph (_see instructions above_).
1. Install the [Sourcegraph Bitbucket Server plugin](../../integration/bitbucket_server.md#sourcegraph-bitbucket-server-plugin) on your Bitbucket Server instance.
1. In Sourcegraph, go to **Site admin > Manage repositories** and edit the Bitbucket Server configuration.
1. Add the `"webhooks"` property to `"plugin"` (you can generate a secret with `openssl rand -hex 32`):<br /> `"plugin": {"webhooks": {"secret": "verylongrandomsecret"}}`
1. Click **Update repositories**.

> 1. Sourcegraph now automatically creates a webhook on your Bitbucket Server instance with the name `sourcegraph-`, followed by the unique ID of your Sourcegraph instance. The webhook is configured for the `pr` and `repo` event types.

Reviewer 2: ```suggestion
1. Sourcegraph now automatically creates a webhook on your Bitbucket Server instance with the name `sourcegraph-`, followed by the unique ID of your Sourcegraph instance. It is configured to deliver `pr` and `repo` events.
```
